### ISSUE TITLE
Error: [@RNC/AsyncStorage]: NativeModule: AsyncStorage is null.

### ISSUE THREAD

**Issue by Anusha-mathur** [react-native-async-storage/async-storage#1117]
### What happened?

![Screenshot_1720692702](https://github.com/react-native-async-storage/async-storage/assets/175305371/14c91740-a231-4c66-b334-0a5f946e1236)

"@react-native-async-storage/async-storage": "~1.23.1",

Most of the solutions given by other people are -- 
- uninstall the app
- restart the metro
- fresh install of dependency
but nothing seems to be working.

This issue is the possible duplicate of 
- https://github.com/react-native-async-storage/async-storage/issues/1105
but this issue is closed due to inactivity.

### Version

0.74.3

### What platforms are you seeing this issue on?

- [X] Android
- [ ] iOS
- [ ] macOS
- [ ] Windows
- [ ] web

### System Information

```Fetching system and libraries information...
System:
  OS: Linux 6.5 Ubuntu 22.04.4 LTS 22.04.4 LTS (Jammy Jellyfish)
  CPU: (12) x64 AMD Ryzen 5 5600H with Radeon Graphics
  Memory: 5.11 GB / 19.38 GB
  Shell:
    version: 5.1.16
    path: /bin/bash
Binaries:
  Node:
    version: 20.11.1
    path: /usr/local/bin/node
  Yarn: Not Found
  npm:
    version: 10.2.4
    path: /usr/local/bin/npm
  Watchman: Not Found
SDKs:
  Android SDK: Not Found
IDEs:
  Android Studio: Not Found
Languages:
  Java:
    version: 17.0.11
    path: /usr/bin/javac
  Ruby: Not Found
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.74.3
    wanted: 0.74.3
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: Not found
  newArchEnabled: false

```

### Steps to Reproduce

Upgraded the react native version from 0.73 -> 0.74
and once i tried to build the app, it is crashing as soon it start.

**Comment by arslanyasin** [comment]
i was getting same issue i just reinstall package and then uninstall app and fresh install works for me

**Comment by aishwarya-art** [comment]
![image](https://github.com/user-attachments/assets/b0c474ea-0848-407d-8d19-a20e111756b8)
I am facing the same issue, tried out the mentioned solutions, doesn't seem to work

**Comment by bernatfortet** [comment]
Same here

**Comment by Anusha-mathur** [comment]
@aishwarya-art @bernatfortet 
You can downgrade the library version to " ~1.17.11" and check if this helps

**Comment by aishwarya-art** [comment]
It didn't work with downgrading the library, any other solution ? Thank you in advance

**Comment by krizzu** [comment]
This issue is mostly due AsyncStorage not being linked on native side. Like @arslanyasin said, reinstalling the application (removing and installing again) should do the trick. If not, please provide a repo where issue can reproduced, so I could help

**Comment by arslanyasin** [comment]
@krizzu @aishwarya-art @bernatfortet i am using version 1.24.0 in other project and it is working fine please check with new version if it works for you

**Comment by bernatfortet** [comment]
I was able to figure it out. This is what I did: (developing for iOS)
- Updated my MacOS to the latest
- Updated XCode to the latest
- Removed/uninstalled the app from the device.
- Then I run this command `npx expo prebuild --clean && npx expo prebuild && cd ios && pod install && cd .. && npx expo run:ios --device`. 

The command above is like a reset + pod installing. It's worked for me as these kinds of errors happen everytime I install a new library that interacts with the native side.

I hope it works!

**Comment by aishwarya-art** [comment]
I have tried the following solutions to resolve issues with linking `@react-native-async-storage/async-storage`:
@krizzu 
1. Uninstalled the app.
2. Restarted Metro.
3. Performed a fresh install of dependencies.

For React Native versions 0.60 and above, native module linking is handled automatically via autolinking, as per the documentation. However, I attempted manual linking with the following steps:

1. Added the dependency in `settings.gradle`:
   ```gradle
   include ':@react-native-async-storage_async-storage'
   project(':@react-native-async-storage_async-storage').projectDir = new File(rootProject.projectDir, '../node_modules/@react-native-async-storage/async-storage/android')
   ```

2. Added the dependency in `app/build.gradle`:
   ```gradle
   dependencies {
       implementation project(':@react-native-async-storage_async-storage')
       // Other dependencies
   }
   ```

3. Updated `MainApplication.java`:
   ```java
   import com.reactnativecommunity.asyncstorage.AsyncStoragePackage; // <-- Add this import

   public class MainApplication extends Application implements ReactApplication {
       @Override
       protected List<ReactPackage> getPackages() {
           @SuppressWarnings("UnnecessaryLocalVariable")
           List<ReactPackage> packages = new PackageList(this).getPackages();
           packages.add(new AsyncStoragePackage()); // <-- Add this line
           return packages;
       }
   }
   ```

4. Ran `./gradlew clean`.

I also checked for errors and warnings using `npx react-native doctor` and verified that there were no issues.
![image](https://github.com/user-attachments/assets/5d9d9be0-c5c5-4b95-b7c5-0877e368c102)

However, nothing seems to be working. Please review the code in the following repository and help resolve the issue, as I am new to this stack:

[Link to Project Code](https://github.com/aishwarya-art/Attendify)

**Comment by krizzu** [comment]
Why try to manual link while you said auto linking works just fine? What's the usecase here?

**Comment by aishwarya-art** [comment]
The issue indicates a problem with the React Native AsyncStorage module not being properly linked, possibly showing an error that AsyncStorage is null. I have tried manual linking to verify if it works.

### Steps Tried:

**1. Uninstall, Rebuild, and Restart the App:**

- Uninstall the app from your device/emulator.
- Rebuild the project:
  ```sh
  npx react-native run-android
  ```
- Restart the app.

**2. Reset Cache:**

- Run the packager with the `--reset-cache` flag:
  ```sh
  npx react-native start --reset-cache
  ```

These steps didn't work for me.

**Comment by krizzu** [comment]
I tried your project and quick investigation showed that it's misconfiguration on `react-native-web` side. I don't know much about it, but I can tell that AsyncStorage and other Native modules are linked properly.

![Screenshot 2024-07-25 at 12 47 22](https://github.com/user-attachments/assets/22111d1d-292b-448f-8037-984b8183bd96)

It just happens that AsyncStorage checks for its native module early (when imported) and throws if it's not there. If you remove this check, however, then you get info that `batched bridge` is missing, so whole native side is not there. Removing react-native-web and fixing babel config launches the app.

For anyone else, please always try AsyncStorage with new project and if you still have issues, please open new issue details and a repository where we can investigate 🙏

**Comment by Samaritan1011001** [comment]
Pinning the version worked for me on RN73.
`"@react-native-async-storage/async-storage": "1.23.1"`

**Comment by tilltu-brian** [comment]
If anyone lands here as they frantically search for a solution, like I did, this was not a simple solution...

My assumptions: because the initial build from EAS did not include AsyncStorage, and I added it later, the native module was not available, and could not become available unless I were to commit my changes and have EAS build from that branch, which was not an option. The solution being to run it locally using the command mentioned above: `npx expo prebuild --clean && npx expo prebuild && cd ios && pod install && cd .. && npx expo run:ios --device`. That did not initially work.

What I had to do to finally get it working:
- `brew install fastlane` if you have not done so already. This is needed for local eas builds.
- `brew install rbenv` if you have not done so already, then `rbenv install 3.3.6` and `rbenv global 3.3.6` to switch to that version of Ruby. The version that comes with MacOS is very outdated. `ruby -v` to verify the version before proceeding.
- `gem update`. You might want, or need, to do `sudo gem uninstall -aIx` and `rm -rf ~/.gem` first
- `brew update` then `brew upgrade cocoapods`
- ensure `pod --version` returns a version and not an error
- then do `npx expo prebuild --clean && npx expo prebuild && cd ios && pod install && cd .. && npx expo run:ios --device`

if, during all that, your ruby version doesn't seem to update, add this to your profile:
```
export PATH="$HOME/.rbenv/bin:$PATH"
eval "$(rbenv init - zsh)"
```
